src/test/ly/stealth/mesos/kafka/SchedulerTest.scala

```diff
@@ -234,4 +234,51 @@ class SchedulerTest extends MesosTestCase {
     assertArrayEquals(Array[AnyRef]("1"), Scheduler.otherTasksAttributes("a").asInstanceOf[Array[AnyRef]])
     assertArrayEquals(Array[AnyRef]("2", "3"), Scheduler.otherTasksAttributes("b").asInstanceOf[Array[AnyRef]])
   }
+
+  @Test
+  def onFrameworkMessage = {
+    val broker0 = Scheduler.cluster.addBroker(new Broker("0"))
+    broker0.active = true
+    val broker1 = Scheduler.cluster.addBroker(new Broker("1"))
+    broker1.active = true
+
+    assertNull(broker0.metrics)
+    assertNull(broker1.metrics)
+
+    val metrics0 = new Broker.Metrics()
+    metrics0.underReplicatedPartitions = 2
+    metrics0.offlinePartitionsCount = 3
+    metrics0.activeControllerCount = 1
+    metrics0.timestamp = System.currentTimeMillis()
+
+    val data = scala.util.parsing.json.JSONObject(Map("metrics" -> metrics0.toJson)).toString().getBytes
+
+    Scheduler.frameworkMessage(schedulerDriver, executorId(Broker.nextExecutorId(broker0)), slaveId(), data)
+
+    // metrics updated for corresponding broker
+    assertNotNull(broker0.metrics)
+    assertNull(broker1.metrics)
+
+    def assertMetricsEquals(expected: Broker.Metrics, actual: Broker.Metrics): Unit = {
```

Comment:
a lot more negative testing and safety here. I think the approach of just these metrics makes sense. If we are only going to use them for health ok but *very fast* folks might get these metrics and start going in and hacking all sorts of different ways to keep exposing more information. I think if we can give them a nice on ramp to "asking the broker" for whatever they wanted (from a metrics perspective) and control that (keeping the ones you have in default that can be added too) rounds out the feature for folks more. We can do it also without "boiling the ocean" and adding all the other refactoring changes but we need a clean way we can reason about, change, support and maintain (in production monitored and running) the pr